src/components/Global/ImageBasic.vue

```diff
@@ -0,0 +1,41 @@
+<script>
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  name: "ImageBasic",
+  props: {
+    src: {
+      type: String,
+      required: true,
+    },
+    alt: {
+      type: String,
+      required: true,
+    },
+    class: {
+      type: String,
+    },
+  },
+  computed: {
+    classes() {
+      return ["image", `${this.class}`];
```

Comment:
I see this pattern everywhere with the classes and I think you should read up again on attribute inheritance - in all of these cases we should remove  the `class` prop and utilize this
https://vuejs.org/guide/components/attrs.html#attribute-inheritance
Likewise all of the props on this component... we could remove and leave them as inheritable attributes 😄

Reply:
The idea with the computed array of classes came from button and links, where I pass prop colorTheme so then need to have computed classes names, so I would return:
classes() {
return [`button ${this.class} button__${this.colorTheme}`];
}
Then I wanted to keep the same schema for classes names in all my components.

So now I conclude I should:
- Use inheritance for classes in general,
- Only when I pass validated props with eg. colorTheme  I can use computed for class name, and it will be added to the classes I pass normally.

Reply:
Pretty much!
Also keep in mind we can keep the "what class would the element have" completely in the template if we want like so:

```vue
<template>
  <button
    :class="[
      'button',
      {
        'button--red': colorTheme === 'red',
        'button--blue': colorTheme === 'blue'
      }
    "]
  >
    <slot />
  </button>
</template>
```

Or make the above a computed as you have said 😄
But generally - unless we desire a different behavior (not inheriting the attributes, or making some very specific element inherit them), we should limit specifying the class (or other attributes) as props as much as possible and utilize what's already built in 😄